pkg/logs/client/tcp/connection_manager.go

```diff
@@ -43,6 +43,10 @@ func NewConnectionManager(endpoint config.Endpoint) *ConnectionManager {
 	}
 }
 
+type timeoutError struct{}
```

Comment:
nits: as this error is tight to tls, can we rename it `tlsTimeoutError` please ?